### PR TITLE
Grids2

### PR DESCRIPTION
**Pull request by spmcbride1201**
Refactored Jumpstate logic into separate files
Reduced renders on Grid components
Fixed Threads and Route components
Lazy load thread data from Finagle
Clean out dead util code.

### CODE REVIEW

File: src/components/JSONComponent.js [hunk 1]
```diff
@@ -19,19 +19,23 @@ class JSONComponent extends Component {
...
   componentWillMount() {
-    this.setState({ headers: Object.keys(this.props.metrics).sort() });
+    this.setState({ headers: this.props.keys.sort() });
   }
+
   componentWillReceiveProps(nextProps) {
-    if (_.isEqual(Object.keys(nextProps.metrics), Object.keys(this.props.metrics))) {
-      this.setState({ headers: Object.keys(nextProps.metrics).sort() });
+    if (!_.isEqual(nextProps.keys, this.props.keys)) {
+      this.setState({ headers: nextProps.keys.sort() });
```
Comment (tilleryd) [inline src/components/JSONComponent.js]: ❤️

File: src/components/JSONComponent.js [hunk 2]
```diff
@@ -58,9 +64,7 @@ class JSONComponent extends Component {
           >
             <Inspector
               data={this.state.headers}
-              onClick={(clicked) => {
-                this.setState({ selectedMetrics: clicked.value });
-              }}
+              onClick={(clicked) => this.setState({ selectedMetrics: clicked.value })}
```
Comment (tilleryd) [inline src/components/JSONComponent.js]: wait, didn't you want to get away from anonymous functions in JSX? 😅

File: src/components/SummaryBar.js [hunk 3]
```diff
@@ -7,12 +7,13 @@ import { getLatestAttribute, getSparkLineOfValue, getSparkLineOfNetChange, parse
 import SummaryBarCard from './SummaryBarCard';
 
 SummaryBar.propTypes = {
-  dashboards: PropTypes.object,
-  interval: PropTypes.number,
-  metrics: PropTypes.object
+  dashboards: PropTypes.object.isRequired,
+  interval: PropTypes.number.isRequired,
+  metrics: PropTypes.object.isRequired,
+  runtime: PropTypes.string
```
Comment (tilleryd) [inline src/components/SummaryBar.js]: I appreciate your commitment to alphabetical ordering of props.

File: src/components/SummaryBar.js [hunk 4]
```diff
@@ -26,26 +27,28 @@ function SummaryBar({ dashboards, metrics, interval }) {
...
-        />
-        {/*<SummaryBarCard
-          chartData={getSparkLineOfValue(metrics, 'jvm/thread/count')}
-          href="/threads"
-          lineOne={getLatestAttribute(metrics, 'jvm/thread/count')}
-          tabIndex={4}
-          title="Threads"
-        />
-        */}  
+        {runtime === "JVM" &&
```
Comment (tilleryd) [inline src/components/SummaryBar.js]: Perhaps there should be separate components per runtime that use SummaryBar.  Per component the SummaryBar will render any number of children.  Then you don't have to use conditionals inside SummaryBar that are specific to a runtime.  Children would get rendered in the same place you have here; between the title div and the dashboard.

File: src/index.js [hunk 5]
```diff
@@ -72,11 +73,12 @@ ReactDOM.render(
...
             path="/summary"
           />
-          {/* <Route
+          {getState().settings.runtime === "JVM" &&
+            <Route
             component={ThreadsGrid}
             path="/threads"
-          />*/  
-          }  
+            />
```
Comment (tilleryd) [inline src/index.js]: indent lines 78 and 79.  you using Prettier?

File: src/store.js [hunk 6]
```diff
@@ -1,94 +1,16 @@
...
-    return { ...state, threadsFilter: payload };
-  },
-  incrementPollingFailures(state, payload) {
-    const pollingFailures = state.pollingFailures + 1;
-    return { ...state, pollingFailures };
-  },
-  resetPollingFailures(state, payload) {
-    return { ...state, pollingFailures: 0 };
-  }
-});
```
Comment (tilleryd) [inline src/store.js]: Nice refactoring of JumpState state functions.

File: src/components/JSONComponent.js (hunk 2 above)
Comment (spmcbride1201) [inline src/components/JSONComponent.js]: lol. Do as I say, not as I do.

File: src/components/SummaryBar.js (hunk 4 above)
Comment (spmcbride1201) [inline src/components/SummaryBar.js]: In my next PR, I've eliminated golang/SummaryBar, but I do still have the runtime gates in place. I think your idea here is the next logical extension of that. I suppose the end goal should be limiting runtime-specific stuff to Container, store, and index. I'm going to open this as a new PR.

File: src/index.js (hunk 5 above)
Comment (spmcbride1201) [inline src/index.js]: Nope, not yet. It's issue #90

Comment (spmcbride1201) [inline src/index.js]: I'm going to fix this indentation issue in my next PR. This seems to be the only thing to fix before merging, and it really doesn't seem worth a rebase.

Comment (spmcbride1201) [inline src/index.js]: Nevermind, fixed it when I fixed the tests.

**Pull request merged**